app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,41 @@
+class FoodsController < ApplicationController
+  before_action :authenticate_user!, except: [:show]
+  before_action :set_food, only: %i[show destroy]
+
+  def index
+    @foods = current_user.foods
+  end
+
+  def create
+    @food = Food.new(food_params)
+    @food.user = current_user
+
+    if @food.save
+      redirect_to foods_path, notice: 'Food item was successfully created.'
+    else
+      flash.now[:alert] = 'Food item could not be saved due to the following errors:'
+      render :new
+      puts 'Not saved'
+      Rails.logger.error("Validation errors: #{@food.errors}")
+    end
+  end
+
+  def destroy
+    if @food.destroy
+      redirect_to foods_path, flash[:notice] = 'Food item was successfully deleted.'
+    else
+      flash.now[:alert] = 'Food item could not be deleted.'
+      render :show
+    end
+  end
```

Comment:
Nice job so far👍🏼. currently, when you try to delete a food, an error is displayed. kindly fix this as this can lead to a poor user experience.
<img width="1256" alt="image" src="https://github.com/Katended/recipe-app/assets/64643995/cca1b533-65f3-42c9-82ec-cf3c0705e0fa">